python/ray/serve/controller.py

```diff
@@ -403,17 +355,17 @@ def _scale_backend_replicas(
                     graceful_timeout_s,
                 ))
 
-    async def _enqueue_pending_scale_changes_loop(self,
-                                                  backend_state: BackendState):
+    def _start_pending_replicas(self):
         for backend_tag, replicas_to_create in self.backend_replicas_to_start.\
                 items():
             for replica_tag in replicas_to_create:
-                replica_handle = await self._start_backend_replica(
-                    backend_state, backend_tag, replica_tag)
+                replica_handle = self._start_backend_replica(
+                    backend_tag, replica_tag)
```

Comment:
Hmm, is this removal of await to allow the start to happen *after* returning from this function, thus removing the need for the following pattern in the controller:
```
_scale_backend_replicas
checkpoint
_enqueue_pending_scale_changes
```

Reply:
The await was removed because we didn't actually need it anymore and this allows it to run in the controller constructor (therefore we don't need to have the annoying logic where we post the `finish_recovering_from_checkpoint` task to the loop).

Reply:
I completely missed that this went from `async` -> `sync`. Makes sense!